checks.d/gearmand.py

```diff
@@ -20,18 +23,36 @@ def _get_client(self,host,port):
         return gearman.GearmanAdminClient(["%s:%s" %
             (host, port)])
 
-    def _get_metrics(self, client, tags):
-        data = client.get_status()
+    def _get_metrics(self, client, specified_tasks, tags):
+        tasks = client.get_status()
         running = 0
         queued = 0
         workers = 0
 
-        for stat in data:
+        if len(specified_tasks) > MAX_NUM_TASKS:
+            raise Exception(
+                "The maximum number of tasks you can specify is {}.".format(MAX_NUM_TASKS))
```

Comment:
Instead of raising an exception, I think we should log a warning and skip the submission of the `*_by_task` metrics and still submit the other metrics.